### PR TITLE
feat: WACI Issuance support

### LINKED ISSUE #553: [WACI-ISSUANCE] Generate OOB Message
The issuer needs to generate a OOB message to be passed to the wallet for Issuance flow

### LINKED ISSUE #554: [WACI-ISSUANCE] Handle Issue Credential - Propose Credential Message
- Map to the original session through parent-thread id
- send issue credential propose offer credential message with following attachments
  - credential manifest 
  - unsigned credential fulfilment

### LINKED ISSUE #555: [WACI-ISSUANCE] Handle Issue Credential - Request Credential Message
- validate credential application data
- send issue credential propose issue credential message with following attachments
  - signed credential fulfilment
  - send web-redirect url

### LINKED ISSUE #556: [WACI-ISSUANCE] QR code support
- Generate QR code from OOB message
- Handle redirection

### PR DESCRIPTION
**Pull request by sudeshrshetty**
- support for WACI issuance
- wallet redirect URL & QR Code support
- integration tests based on aries universal wallet
- Closes #556
- Closes #554
- Closes #553
- Closes #555

Signed-off-by: sudesh.shetty <sudesh.shetty@securekey.com>

### CODE REVIEW

File: cmd/adapter-rest/go.mod [hunk 1]
```diff
@@ -24,3 +24,5 @@ require (
 )
 
 replace github.com/trustbloc/edge-adapter => ../..
+
+replace github.com/hyperledger/aries-framework-go => github.com/sudeshrshetty/aries-framework-go v0.0.0-20220105232448-180bc16fba29
```
Comment (rolsonquadras) [inline cmd/adapter-rest/go.mod:28]: missed to remove this line ?

File: test/bdd/features/issuer_e2e.feature [hunk 2]
```diff
@@ -30,9 +30,9 @@ Feature: Issuer Adapter e2e
...
-      | creditCard      | CreditCard Issuer                     | http://mock-issuer.com:9080/creditCard         | https://trustbloc.github.io/context/vc/examples/credit-card-v1.jsonld                | false                   | CreditCardStatement     |                   |
-      | driversLicense  | Drivers License wth Evidence Issuer   | http://mock-issuer.com:9080/driversLicense     | https://trustbloc.github.io/context/vc/examples/driver-license-evidence-v1.jsonld    | true                    | DrivingLicenseEvidence  | mDL               |
+      | prCard5          | PRCard Issuer                         | http://mock-issuer.com:9080/prCard             | https://trustbloc.github.io/context/vc/examples/citizenship-v1.jsonld                | false                   | PermanentResidentCard   |                   |
+#      | creditCard      | CreditCard Issuer                     | http://mock-issuer.com:9080/creditCard         | https://trustbloc.github.io/context/vc/examples/credit-card-v1.jsonld                | false                   | CreditCardStatement     |                   |
```
Comment (rolsonquadras) [inline test/bdd/features/issuer_e2e.feature:34]: Should these be uncommented ?

File: test/bdd/go.mod [hunk 3]
```diff
@@ -26,3 +26,5 @@ require (
 )
 
 replace github.com/trustbloc/edge-adapter => ../..
+
+replace github.com/hyperledger/aries-framework-go => github.com/sudeshrshetty/aries-framework-go v0.0.0-20220105232448-180bc16fba29
```
Comment (rolsonquadras) [inline test/bdd/go.mod:30]: same as https://github.com/trustbloc/adapter/pull/565/files#r780486032

File: cmd/adapter-rest/go.mod (hunk 1 above)
Comment (sudeshrshetty) [reply cmd/adapter-rest/go.mod]: fixed

File: test/bdd/features/issuer_e2e.feature (hunk 2 above)
Comment (sudeshrshetty) [reply test/bdd/features/issuer_e2e.feature:34]: yes, fixed

File: test/bdd/go.mod (hunk 3 above)
Comment (sudeshrshetty) [reply test/bdd/go.mod]: fixed

**Review by rolsonquadras — APPROVED** [review]

**Review by fqutishat — APPROVED** [review]

**Review by birtony — COMMENTED** [review]
@sudeshrshetty can you add a screenshot of the UI, please?

**Comment by sudeshrshetty** [comment]
> @sudeshrshetty can you add a screenshot of the UI, please?

This PR is about adapter backend integration only, no changes has been done for UI except showing QRCode & Redirect.
The UI changed you see are just JS refactoring to support both CHAPI & WACI and copy of QRcode logic from existing WACI share page.

image

**Pull request merged**